File: src/main/java/dark/leech/text/plugin/js/loader/ConsoleApi.java
package dark.leech.text.plugin.js.loader;

import dark.leech.text.action.Log;

/**
 * Console API for JavaScript plugin scripts. Matches vBooks JSLog behavior - provides log() method
 * for plugin debugging.
 */
public class ConsoleApi {

    /**
     * Log a message to the application log. Called from JavaScript as console.log() or
     * Console.log()
     *
     * @param msg The message to log
     */
    public void log(Object msg) {
        if (msg != null) {
            String msgStr = JSResponse.getString(msg);
            if (msgStr != null) {
                Log.add(msgStr);
            }
        }
    }
}

File: engine/src/commonMain/kotlin/dev/haipham22/leechtext/plugin/js/loader/ConsoleApi.kt
package dev.haipham22.leechtext.plugin.js.loader

import dev.haipham22.leechtext.log.EngineLogger

/**
 * Console API cho plugin scripts (port từ ConsoleApi.java — match vBooks JSLog).
 */
class ConsoleApi(
    private val log: EngineLogger,
) {
    fun log(msg: Any?) {
        if (msg != null) {
            JSResponse.getString(msg)?.let { log.add(it) }
        }
    }
}
